### PR TITLE
[SPARK-20930][ML]  Destroy broadcasted centers after computing cost in KMeans

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
 Destroy broadcasted centers after computing cost
## How was this patch tested?
existing tests

### CODE REVIEW

**Comment by srowen** [comment]
Yes, I agree with that. It looks like there may be a similar instance in `GradientDescent.scala`, where `bcWeights` is used in a `treeAggregate` but then never destroyed.

In every other similar instance I see, yes, the broadcast is destroyed after it's used to collect a result.

**Comment by zhengruifeng** [comment]
@srowen  Thanks for pointing that out. Btw, I reviewed all calls of broadcast in ml again and found this instance also happens in `LDAModel.scala`. 
I think now all broadcasted objects in ml will be destroyed after computation, except transform/predict/predictSoft/etc.

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala [hunk 1]
```diff
@@ -320,6 +320,7 @@ class LocalLDAModel private[spark] (
 
         docBound
       }.sum()
+    ElogbetaBc.destroy(blocking = false)
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: Looks good; how about in the `getTopicDistributionMethod` method too? I actually think that broadcast is pointless.

Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: `getTopicDistributionMethod` method is only used in `ml.clustering.LDA#transform`, so I think this maybe useful if the model size is large.

Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: @zhengruifeng but the broadcast isn't actually used. Its `.value` is called, locally, not from a distributed method.

Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: @srowen You are right. I removed the unnecessary broadcasting.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**